backend/src/Core/FurnitureShop.Core.Contracts/Dtos/ProductsDTOs.cs

```diff
@@ -1,32 +1,44 @@
 using System;
 using System.Collections.Generic;
 
-namespace FurnitureShop.Core.Contracts.Mobile.Products
+namespace FurnitureShop.Core.Contracts.Dtos
 {
-    public class ProductDetailsDTO : ProductInfoDTO
-    {
-        public string Description { get; set; }
-        public string? ModelUrl { get; set; }
-    }
-    public class ProductInfoDTO
+    public class ProductDTOBase
     {
         public string Name { get; set; }
         public double Price { get; set; }
         public double? AverageRating { get; set; }
         public string? PreviewPhotoURL { get; set; }
-        public bool InFavourites { get; set; }
-        public bool InShoppingCart { get; set; }
         public Guid? CategoryId { get; set; }
     }
+    public class ProductWithDetailsDTOBase : ProductDTOBase
+    {
+        public string Description { get; set; }
+        public string? ModelUrl { get; set; }
+    }
 
-    public class ProductWithDetailsDTO
+    public class ProductWithDetailsDTO : ProductWithDetailsDTOBase
     {
         public Guid Id { get; set; }
-        public ProductDetailsDTO ProductDetails { get; set; }
     }
-    public class ProductDTO
+    public class ProductDTO : ProductDTOBase
     {
         public Guid Id { get; set; }
-        public ProductInfoDTO ProductInfo { get; set; }
+    }
+    public class ProductUDTO : ProductDTO
```

Comment:
U can have ProductDTO in Mobile and Web folders with different namespace. And in Handler u can use aliases(whole namespace path) to specify which object will be returned

Reply:
IMO, u need to inject Dtos folder to each folder(Mobile/Web/Shared) respectively